From 8f78673a42962c52cd1a458239a0ac81bdb00939 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@nabladev.com>
Date: Mon, 7 Jul 2025 22:00:30 +0200
Subject: [PATCH 953/954] fix(stm32mp2): enable VDDIO3SV and VDDIO4SV

Enable VDDIO3SV and VDDIO4SV to permit access to I2C4 pins, which
are mandatory to configure PMIC.

Upstream-Status: Pending
Signed-off-by: Marek Vasut <marex@nabladev.com>
---
 plat/st/stm32mp2/bl2_plat_setup.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/plat/st/stm32mp2/bl2_plat_setup.c b/plat/st/stm32mp2/bl2_plat_setup.c
index c60463f475..efdfb312b0 100644
--- a/plat/st/stm32mp2/bl2_plat_setup.c
+++ b/plat/st/stm32mp2/bl2_plat_setup.c
@@ -480,6 +480,9 @@ void bl2_el3_plat_arch_setup(void)
 
 	clk_fixed_register(fdt);
 
+/* Enable VDDIO3SV and VDDIO4SV */
+mmio_setbits_32(0x44210000, 0x300);
+
 	if (stm32mp2_clk_init(fdt) < 0) {
 		panic();
 	}
-- 
2.53.0

